src/core/mod.rs

```diff
@@ -0,0 +1,30 @@
+mod arbitrary;
+mod container;
+mod fmt;
+mod multiops;
+mod proptests;
+mod store;
+
+// Order of these modules matters as it determines the `impl` blocks order in
+// the docs
+mod cmp;
+mod inherent;
+mod iter;
+mod ops;
+
+#[cfg(feature = "serde")]
+mod serde;
+
+mod serialization;
+
+use self::cmp::Pairs;
+use crate::Value;
+
+pub use self::iter::IntoIter;
+pub use self::iter::Iter;
+
+/// A generic compressed bitmap using the [Roaring bitmap compression scheme](https://roaringbitmap.org/).
+#[derive(PartialEq)]
+pub struct RoaringBitmap<V: Value> {
+    containers: Vec<container::Container<V>>,
```

Comment:
I worry that using a Vec for RoaringBitmap64 might not be the best idea. For RoaringBitmap32, in the worst case, we're moving around 64k containers, but for RoaringBitmap64, inserting a new container may involve quite a lot of copying, which is why the original used a BTreeMap instead.

Reply:
I guess that's depends on the distribution of your dataset/workload, using a `BTreeMap` isn't great either in term locality/pointer chasing.

I would say that for most of the case it should be fine, at least [the paper](https://r-libre.teluq.ca/930/1/Roaring64bits.pdf) didn't notice large difference in memory usage.

But maybe both approach should be offered?